File: API/Models/Product.cs


using API.Database;

namespace API.Models;

[DbTable("dbo.product")]
public class Product
{
    [DbPrimaryKey]
    public int ProductId { get; set; }

    [DbColumn]
    public string Name { get; set; }
    
    [DbColumn]
    public string Slug { get; set; }
    
    [DbColumn]
    public string Description { get; set; }
    
    [DbColumn]
    public int ProductTypeId { get; set; }
    
    [DbColumn]
    public bool IsTradeable { get; set; }
    
    [DbColumn]
    public bool IsNew { get; set; }
    
    [DbColumn]
    public bool IsPromotional { get; set; }
    
    [DbColumn]
    public bool IsExclusive { get; set; }
    
    [DbColumn]
    public string Sku { get; set; }
    
    [DbColumn]
    public int PriceTypeId { get; set; }
    
    [DbColumn]
    public decimal Price { get; set; }
    
    [DbColumn]
    public decimal PremiumPrice { get; set; }

    [DbColumn]
    public DateTime CreatedAt { get; set; }

    [DbColumn]
    public DateTime UpdatedAt { get; set; }
}

File: client/src/types/product.ts
import type { Tag } from "./metadata";
import type { Subcategory } from "./subcategory";

export interface Product {
  productId: number;
  name: string;
  slug: string;
  productTypeId: number;
  isTradeable: boolean;
  isNew: boolean;
  isPromotional: boolean;
  isExclusive: boolean;
  isDemoProduct: boolean;
  sku: string;
  price: number;
  premiumPrice: number;
  priceIcon: string;
  primaryImage: ProductImage | null;
  updatedAt: string | null;
}

export interface ProductImage {
  productImageId: number;
  productId: number;
  imageUrl: string;
  altText?: string;
  isPrimary: boolean;
  displayOrder: number;
}

export interface ProductDetail extends Product {
  description?: string;
  priceTypeId: number;
  images: ProductImage[];
  subcategories: Subcategory[];
  tags: Tag[];
  updatedAt: string | null;
}

export interface ProductFormRequest {
  name: string;
  slug: string;
  productTypeId: number;
  description: string | null;
  isTradeable: boolean;
  isNew: boolean;
  isPromotional: boolean;
  isExclusive: boolean;
  price: number;
  premiumPrice: number;
  priceTypeId: number;
  subcategoryIds: number[];
  tags: string[];
  updatedAt: string | null;
}

export interface AddProductImageRequest {
  file: File;
  altText: string | null;
  setAsPrimary: boolean;
}